mlflow/openai/_openai_autolog.py

```diff
@@ -161,12 +177,18 @@ def patched_call(original, self, *args, **kwargs):
                 _logger.warning(f"Encountered unexpected error when ending trace: {inner_e}")
         raise e
 
+    if not isinstance(raw_result, (BaseModel, Stream)):
```

Comment:
Can we check if the response type is `LegacyAPIResponse`? OpenAI may adds new response type, which should not be processed by `try_parse_raw_response`.

Reply:
We could add `from openai._legacy_response import LegacyAPIResponse` but that would another risk (an import from a private module)

Reply:
What about changing the logic to `if hasattr(raw_result, "parse")`? We may be able to avoid being bothered by the internal interface change of the SDK?